toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,319 @@
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		ifaceHardwareAddr := strings.TrimSpace(iface.HardwareAddr.String())
+		if networkData.Device == iface.Name || strings.EqualFold(ifaceHardwareAddr, networkData.Device) {
+			deviceName = iface.Name
+			return
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	var networkSection strings.Builder
+
+	// Write network section tag
+	networkSection.WriteString("[Network]\n")
+
+	// Currently only supports static IP setting
+	// Write IP and netmask
+	if networkData.NetMask != "" && networkData.Ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrNumber, _ := stringMask.Size()
+		networkSection.WriteString(fmt.Sprintf("Address=%s/%d\n", networkData.Ip, cidrNumber))
+	}
+
+	// Write Gateway
+	if networkData.GateWay != "" {
+		networkSection.WriteString(fmt.Sprintf("Gateway=%s\n", networkData.GateWay))
+	}
+
+	// Write Nameserver
+	for _, nameserver := range networkData.NameServers {
+		networkSection.WriteString(fmt.Sprintf("DNS=%s\n", nameserver))
+	}
+
+	err = file.Append(networkSection.String(), fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network"
+		filePrefix     = "10"
+		squashErrors   = false
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	logger.Log.Debugf("Start creating network file")
+
+	networkFilePath := fmt.Sprintf("%s/%s-%s-%s.network", networkFileDir, filePrefix, networkData.BootProto, deviceName)
+	exists, err := file.PathExists(networkFilePath)
+	if exists && err == nil {
+		return fmt.Errorf("network file (%s) already exists", networkFilePath)
+	}
+
+	err = file.Create(networkFilePath, 0644)
+	if err != nil {
+		logger.Log.Errorf("Error creating file %s: %s", networkFilePath, err)
+		return
+	}
+
+	defer func() {
+		// Delete the network file on failure
+		if err != nil {
+			err = shell.ExecuteLive(squashErrors, "rm", networkFilePath)
```

Comment:
>err = shell.ExecuteLive(squashErrors, "rm", networkFilePath) [](http://example.com/codeflow?start=3&length=60)

Let's use Go library functions wherever possible instead of spinning a new process. In this case, I believe we've been using `os.Remove`.

Reply:
Fixed